magento/magento2
# Fixes issue 23411 - Remove misplaced critical log "No such entity with customerId = xx"

### Description (*)
This is a new attempt at fixing https://github.com/magento/magento2/issues/23411, this is based on https://github.com/magento/magento2/pull/23952

Review comments from @ihor-sviziev have been taken into account.
The removal of the comment above the constant `CHECKOUT_STATE_BEGIN` was done to satisfy Magento's coding standard checks, it added no useful information.

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/23411: After upgrading to new version - No such entity with customerId critical logs
2. https://github.com/magento/magento2/issues/24009: New Quote is not creating after order place for logged in customer


### Manual testing scenarios (*)
See: https://github.com/magento/magento2/pull/23952 or https://github.com/magento/magento2/issues/23411#issuecomment-506961956

### Questions or comments


### Contribution checklist (*)
